runtime/src/main/java/org/corfudb/runtime/view/LayoutManagementView.java

```diff
@@ -110,6 +111,43 @@ public void bootstrapNewNode(String endpoint)
         log.info("bootstrapNewNode: New node {} bootstrapped.", endpoint);
     }
 
+    /**
+     * Removes a node from the layout
+     * @param currentLayout the layout to remove the node from
+     * @param endpoint the node to remove
+     */
+    public void removeNode(Layout currentLayout, String endpoint) throws QuorumUnreachableException,
+            CloneNotSupportedException,
+            LayoutModificationException, OutrankedException {
+
+        LayoutBuilder builder = new LayoutBuilder(currentLayout);
+        Layout newLayout = builder.removeLayoutServer(endpoint)
+                .removeSequencerServer(endpoint)
+                .removeLogunitServer(endpoint)
+                .removeUnResponsiveServer(endpoint)
+                .setEpoch(currentLayout.getEpoch() + 1)
+                .build();
+
+        // Seal after constructing the layout, so that the system
+        // isn't blocked if the builder throws an exception
+        currentLayout.setRuntime(runtime);
+        sealEpoch(currentLayout);
+
+        newLayout.setRuntime(runtime);
+        attemptConsensus(newLayout);
+
+        runtime.invalidateLayout();
+
+        try {
+            reconfigureSequencerServers(currentLayout, newLayout, false);
+        } catch (InterruptedException ie) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)

Reply:
Done.